lib/pbench/server/__init__.py

```diff
@@ -191,15 +194,39 @@ def max_retention_period(self) -> int:
         allows a dataset to be retained.
 
         Returns:
-            integer representing retention period in days
+            An integer number of days representing the maximum retention period.
         """
-        try:
-            retention_days = int(
-                self._get_conf("pbench-server", "maximum-dataset-retention-days")
-            )
-        except (NoOptionError, NoSectionError):
-            retention_days = self.MAXIMUM_RETENTION_DAYS
-        return retention_days
+        return self.conf.getint(
+            "pbench-server",
+            "maximum-dataset-retention-days",
+            fallback=self.MAXIMUM_RETENTION_DAYS,
+        )
+
+    @property
+    def default_retention_period(self) -> int:
+        """Produce an integer representing the default number of days the server
+        allows a dataset to be retained.
+
+        Returns:
+            An integer number of days representing the default retention period.
+        """
+        return self.conf.getint(
+            "pbench-server",
+            "default-dataset-retention-days",
+            fallback=self.DEFAULT_RETENTION_DAYS,
+        )
```

Comment:
So what does `ConfigParser.getint` do if the string in the config file isn't an integer? The old code specifically handled an exception and generated a meaningful cleanup exception for the upload error logic. The new code will result in a fallback generic `Exception` handler, which is less desirable. (Although it should still be cleaned up.)

Interestingly, the `ConfigParser` documentation for `getboolean` specifically references a `ValueError` if it's not an expected value, while `getint` and `getfloat` give no clue what'll happen.

Reply:
First, here is the behavior of `getint`:

```
$ python3
Python 3.10.7 (main, Sep  7 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import configparser
>>> cp = configparser.ConfigParser()
>>> cp['server'] = {'is_an_int': 42, 'is_not_an_int': 'forty-two'}
>>> cp.getint('server', 'is_an_int')
42
>>> cp.getint('server', 'is_not_an_int')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/configparser.py", line 819, in getint
    return self._get_conv(section, option, int, raw=raw, vars=vars,
  File "/usr/lib64/python3.10/configparser.py", line 809, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
  File "/usr/lib64/python3.10/configparser.py", line 804, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: invalid literal for int() with base 10: 'forty-two'
>>> cp.getint('server', 'is_not_an_int', fallback=43)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.10/configparser.py", line 819, in getint
    return self._get_conv(section, option, int, raw=raw, vars=vars,
  File "/usr/lib64/python3.10/configparser.py", line 809, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
  File "/usr/lib64/python3.10/configparser.py", line 804, in _get
    return conv(self.get(section, option, **kwargs))
ValueError: invalid literal for int() with base 10: 'forty-two'
```

Reply:
I have made a change to restore the old cleanup behavior.